Adds support for Python 3.3, 3.4, and 3.6

jfinkels opened the pull request:
This is a unit test that currently fails due to issue #5. This pull request is based on the commit from pull request #8.

jfinkels commented:
I am not familiar with the testing environment, where should I add `astunparse` as a testing dependency?

serge-sans-paille commented:
On Sun, Apr 23, 2017 at 07:37:04PM -0700, jfinkels wrote:
> I am not familiar with the testing environment, where should I add `astunparse` as a testing dependency?

in ``test_require``, see

http://setuptools.readthedocs.io/en/latest/setuptools.html

jfinkels commented:
Okay, I got the tests to run on Travis (by adding the new test dependency `astunparse` to `tox.ini`). I have added code to workaround the issue with Python 3.3 and 3.4. With these changes, the new `test_unparse` unit test that I've added seems to pass! However, I seem to have introduced a regression in the `test_compile` unit test and I'm not sure how to determine where the issue is:

```sh
___________________________ SelfTestCase.testCompile ___________________________

self = <tests.test_self.SelfTestCase testMethod=testCompile>

    def testCompile(self):
        for src_py in self.srcs:
            with open(src_py) as f:
                content = f.read()
            gnode = gast.parse(content)
>           compile(gast.gast_to_ast(gnode), src_py, 'exec')
E           TypeError: required field "lineno" missing from expr

tests/test_self.py:27: TypeError
```

pbrunet commented on gast/ast3.py:
```diff
@@ -88,6 +88,32 @@ def visit_ExceptHandler(self, node):
+                args=self._visit(node.args),
+                keywords=self._visit(node.keywords),
+                starargs=[],
+                kwargs=[],
```
Here it is not a star argument but an optional argument so empty should be "None" instead of "[]". This is was generated your error on travis

jfinkels commented:
@pbrunet Merci! The tests now pass :)

serge-sans-paille commented:
Apart from the remark on #8, i'm ok with this one

jfinkels commented:
In pull request #8, @serge-sans-paille wrote

> I'm okay with this PR, but I'd like to make the stuptools dependencies optional. If you could
>
> ```python
> try:
>    from setuptools import setup
>    kwargs = {...}
> except ImportError:
>    from distutils.core import setup
>    kwargs = {}
>
> setup(..., **kwargs)
> ```
> that would be great

I will make this change, no problem. I'm just curious, do you work in an environment that doesn't have setuptools available?

jfinkels commented:
Instead of doing the conditional import as you suggested, I left the `from distutils.core import setup` line as-is, added the `tests_require` and `test_suite` keyword arguments, and it seems to work as I expect.

serge-sans-paille commented:
> I will make this change, no problem. I'm just curious, do you work in an environment that doesn't have setuptools available?

When I'm working with virtual env, no; When I start a debian from scratch (which does not happen that often) yes.

serge-sans-paille commented on tests/test_self.py:
```diff
@@ -1,28 +1,38 @@
 
+    def setUp(self):
+        self.srcs = glob.glob(os.path.join(gast.__path__[0], '*.py'))
+
```
great code syndication.

serge-sans-paille commented:
Thanks!

jfinkels commented:
@serge-sans-paille can you publish these changes on PyPI when you have some free time? Thanks!

serge-sans-paille commented:
Done! cc @jfinkels @mattbar
